tox.ini

```diff
@@ -22,6 +22,14 @@ deps =
     zope.proxy
     zope.security
 
+[testenv:pypy3]
+# On certain platforms, at least OS X, pypy3 does not correctly read
+# a LANG environment variable; when cpython3.2 would return 'utf-8'
+# from locale.getpreferredencoding(), pypy3 returns 'us-ascii', causing
+# two test failures. Fix that by setting something it does read.
```

Comment:
In a slightly related issue, the documentation for `locale.getpreferredencoding` states that it's not thread safe. There are three implementations. One of them, under certain unspecified Unix platforms contains the following code (for non-Windows)(simplified):
```python
        def getpreferredencoding(do_setlocale = True):
            if do_setlocale:
                oldloc = setlocale(LC_CTYPE)
                try:
                    setlocale(LC_CTYPE, "")
                except Error:
                    pass
                result = nl_langinfo(CODESET)
                ...
                setlocale(LC_CTYPE, oldloc)
            else:
                ...
        return result
```
That looks pretty non-thread safe to me! Which in turn makes `cgi:FieldStorage` and `zope.publisher.browser:ZopeFieldStorage` non-thread safe, as well as `zope.publisher.browser:BrowserRequest.processInputs` which uses `ZopeFieldStorage`. 

Maybe we're getting lucky and all supported platforms use the alternate implementation that goes through `getdefaultlocale`.